core/model/modx/processors/security/login.class.php

```diff
@@ -140,23 +140,29 @@ public function checkIsBlocked() {
         /** @var modUserProfile $profile */
         $profile = $this->user->Profile;
 
-        if ($profile->get('failed_logins') >= $this->modx->getOption('failed_login_attempts') &&
-            $profile->get('blockeduntil') > time()) {
-            return $this->modx->lexicon('login_blocked_too_many_attempts');
-        }
-
-        if ($profile->get('failedlogincount') >= $this->modx->getOption('failed_login_attempts')) {
-            $profile->set('failedlogincount', 0);
-            $profile->set('blocked', 1);
-            $profile->set('blockeduntil', time() + (60 * $this->modx->getOption('blocked_minutes')));
+        /* Update block state */
+        if ($profile->get('blockeduntil') < time()) {
+            if ($profile->get('blockeduntil') > 0) {
+                $profile->set('failedlogincount', 0);
+            }
+            $profile->set('blockeduntil', 0);
             $profile->save();
         }
-        if ($profile->get('blockeduntil') != 0 && $profile->get('blockeduntil') < time()) {
+        if ($this->user->passwordMatches($this->givenPassword)) {
```

Comment:
This seems a bit odd to be added to a checkIsBlocked method - is it necessary to check the password here again?

Reply:
More than odd; would this not destroy the ability to use external authentication via plugins?

Reply:
No, it will not, it only resets "failedlogincount" and has nothing to do with the evaluation of blocked state. 
A successful login should reset "failedlogincount". Using an external authentication plugin will still pass, yet "failedlogincount" will be reset time based on next login.

Reply:
With an external authentication plugin, the password will not be stored locally. So this does not seem like a valid solution. I will need to dig into and test this with a system using external authentication, but I can almost guarantee this will not work.

Reply:
Wouldn't it make more sense to reset the `failedlogincount` in `afterLogin()` instead of doing so in `checkIsBlocked()`?

Reply:
@JoshuaLuckers, A successful login should reset `failedlogincount` **before** `checkIsBlocked()`, so it would bypass the `failedlogincount` block status _(as the user should not be blocked anymore based on the successful login)_.
@opengeek 's assumption is **only** valid if a user is already blocked because of `failedlogincount`, then successfully logs in via an external authentication plugin _(excluding Login extra)_. Only in such case, as I replied earlier, it will not reset the count and will be time based. Thus, it's a specific use case, based on the assumption of that (authentication plugin) incrementing  `failedlogincount` manually, as it's only incremented via the password validation (refer to `checkPassword()` logic).